src/varity/vcf_to_hgvs/protein.clj

```diff
@@ -308,12 +308,14 @@
                      format-alt-prot-seq
                      (subs (dec (+ ppos offset)))
                      (string/index-of "*"))]
-    (mut/protein-frame-shift (mut/->long-amino-acid ref)
-                             (coord/protein-coordinate (+ ppos offset))
-                             (mut/->long-amino-acid alt)
-                             (if (and ter-site (pos? ter-site))
-                               (coord/protein-coordinate (inc ter-site))
-                               (coord/unknown-coordinate)))))
+    (if (= alt \*)
+      (protein-substitution (+ ppos offset) (str ref) (str alt)) ; eventually fs-ter-substitution
```

Comment:
:memo: In certain cases, frameshift would be fixed as `fs-ter-substitution` by [the 3'-rule application to protein sequence](https://github.com/chrovis/varity/blob/e51d56c8384e4050603f7036d8a03bf31878e4fe/src/varity/vcf_to_hgvs/protein.clj#L344). So I have added the `fs-ter-substition` check here.

Reply:
Indeed, such cases potentially occur as you said. It is extremely rare.

Do you have any case of that? I would like you to add the test case if you already have.

No problem even if you do not have the case. I will approve this PR even then because I think it is a bit difficult to prepare the test case.

Reply:
Sorry, the variant query I attempted appears no longer comes in this path by the `backward-shift` fix. :thinking: 

Instead, I have added test cases for `varity.vcf-to-hgvs.protein/mutation` in https://github.com/chrovis/varity/pull/75/commits/6b2ef4ce48fdb22573615d8431cf76c2d328017f, which includes inputs before and after the 3'-rule of coding DNA. The 3'-rule for protein sequence and `fs-ter-substitution` check work correctly in both cases.

Reply:
Okay, thank you very much. The test case using `varity.vcf-to-hgvs.protein/mutation` looks good to me.